lib/scrapPaperReader.sql

```diff
@@ -0,0 +1,34 @@
+-- BLOCK get_barcode_metadata
+SELECT
+    s.id AS submission_id,
+    iq.assessment_instance_id AS assessment_instance_id,
+    iq.id AS instance_question_id,
+    s.submitted_answer->>'_pdf_barcode_scan' AS barcode,
+    f.id AS file_id
+FROM 
+    submissions AS s
+    JOIN variants AS v ON (v.id = s.variant_id)
+    LEFT JOIN instance_questions AS iq ON (iq.id = v.instance_question_id)  -- LEFT JOIN HERE SO INSTRUCTOR CAN USE THIS FEATURE TOO
+    LEFT JOIN files AS f ON (f.instance_question_id = iq.id)
+WHERE $match AND type = 'pdf_barcode_scan';
+
+-- BLOCK update_barcodes
+DROP TABLE IF EXISTS barcode_file_ids_temp;
+
+CREATE TEMP TABLE barcode_file_ids_temp
+    (file_id BIGINT NOT NULL PRIMARY KEY, barcode TEXT);
+
+INSERT INTO
+    barcode_file_ids_temp(file_id, barcode)
```

Comment:
I couldn't find a clean example of how to do an upsert. This is the most efficient way I could think of.

Reply:
`INSERT INTO ... ON CONFLICT ...` is going to be your best bet, that's the pattern we use throughout PL. See [here in `lib/config.sql`](https://github.com/PrairieLearn/PrairieLearn/blob/020e3097723a6b72f8079d333bc666ff0c8bd268/lib/config.sql#L1) for an example.